src/Illuminate/Foundation/Console/RouteCacheCommand.php

```diff
@@ -30,13 +45,15 @@ class RouteCacheCommand extends Command {
 	/**
 	 * Create a new route command instance.
 	 *
-	 * @param  \Illuminate\Filesystem\Filesystem  $files
-	 * @return void
+	 * @param  \Illuminate\Filesystem\Filesystem $files
+	 * @param  \Illuminate\Routing\Router  $router
 	 */
-	public function __construct(Filesystem $files)
+	public function __construct(Filesystem $files, Router $router)
 	{
 		parent::__construct();
 
+		$this->router = $router;
+		$this->routes = $router->getRoutes();
```

Comment:
> Unfortunately that isn't working right.

Well, this break everyone else code that using `php artisan route:cache` including myself.